### DISCUSSION TITLE
Improvements : Allo v2.1

### DISCUSSION THREAD

**Discussion by thelostone-mc** [allo-protocol/allo-v2#526] [General]
## Contracts
- msg.value replaced with _msgValue + msg.sender replaced with _msgSender
- reimplement batchAllocate to support native batch allocation across rounds
- [allow all strategies to be cloneable](https://hackmd.io/@carlb/H194byEi6?comment=003d83ad-0249-4630-8f6b-6e6dd8da4fc2&utm_source=comment-card&utm_medium=icon)
- Ensure default receive is virtual and can be called only by Allo
- Move `receive()` function over to BaseStrategy

## Misc
- Move ownership back to deployer wallet
- [Document JSON format for pool and profile](https://hackmd.io/@carlb/H194byEi6?comment=90d0cd25-0f4a-4bb3-8a3e-02664b8fc2ce&utm_source=comment-card&utm_medium=icon)

## SDK
- allo sdk can send the transactions also (could be good to keep the current impl of just encodingFunctionData also)
- listen for emitted events and return the data being waited for (like the above example)
- ReactQuery wrapper for hooks (allo.createPool.useMutation(params, { onSuccess, onError })

## Documenation
- Quick Start Tutorials
- Explaining the Allo Vocabulary

### On-Going Discussion
- [Remove the requirement of a Profile when creating Pools](https://hackmd.io/@carlb/H194byEi6?comment=d3b61af7-76c8-4016-807a-fe9e42f0ff08&utm_source=comment-card&utm_medium=icon)
- [EIP-712 signed authorization](https://github.com/allo-protocol/allo-v2/discussions/483)

**Comment by carlbarrdahl** [comment]
Regarding ReactQuery, this might be helpful as a guide to writing a wrapper:
https://github.com/ts-rest/ts-rest/blob/main/libs/ts-rest/react-query/src/lib/react-query.ts
